core/templates/domain/admin/admin-backend-api.service.spec.ts

```diff
@@ -1283,6 +1309,65 @@ describe('Admin backend api service', () => {
     expect(failHandler).not.toHaveBeenCalled();
   }));
 
+  it('should generate dummy suggestion questions', fakeAsync(() => {
+    let action = 'generate_dummy_suggestion_questions';
+    let skillId = 'dc3k2ldd';
+    let numberOfQuestions = 2;
+    let payload = {
+      action: action,
+      skill_id: skillId,
+      num_dummy_suggestion_ques_generate: numberOfQuestions,
+    };
+
+    abas
+      .generateDummySuggestionQuestionsAsync(skillId, numberOfQuestions)
+      .then(successHandler, failHandler);
+
+    let req = httpTestingController.expectOne('/adminhandler');
+    expect(req.request.method).toEqual('POST');
+    expect(req.request.body).toEqual(payload);
+    req.flush(200);
+    flushMicrotasks();
+
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+  }));
+
+  it(
```

Comment:
More tests is not always better, you are already editing a file of 1.3k LOCs. In this case, you have a function with 3 lines of code, all of which are trivial. The test above will suffice, you can delete this one

Reply:
I'm not so sure about this. I agree that more tests are not always better (i.e if the extra tests are exercising the same feature as another existing test). However if each test is exercising a unique feature about the component/service, then dropping one of the tests removes our source of feedback of whether the feature works as intended or not. I think both tests exercise unique features: one test for the service to handle successful question suggestion generation, and the other test for the service to handle server errors raised from trying to generate question suggestions.

If the concern is that this test file is becoming too long, then I think it could be worth looking through both this test file and the service file to determine if we can separate these features into multiple separate services.

Also, I'm not sure which function has 3 lines of code. Are we referring to `generateDummySuggestionQuestionsAsync()`? If so, then I don't think these lines of code are trivial, as they determine the body of the network request that carries out the user's intent of generating dummy question suggestions.

More importantly however, frontend tests document what we expect the component/service to do. The tests don't care about how the code is implemented as long as it achieves our outlined expectations. Whether it takes 1, 3, 10 or 100 lines of code to pass should not matter to the test.

Reply:
🤷